src/files.js

```diff
@@ -74,7 +80,13 @@ function doUpload(cozy, data, method, path, options) {
   if (lastModifiedDate) headers['Date'] = lastModifiedDate.toGMTString()
   if (ifMatch) headers['If-Match'] = ifMatch
 
-  return cozyFetch(cozy, path, {
+  let finalpath = path
+  if (metadata) {
+    const metadataId = await sendMetadata(cozy, metadata)
+    if (metadataId) finalpath += `&MetadataID=${metadataId}`
```

Comment:
Are we sure that the path always contains a query-string?

Reply:
You are right, if doUpload is called by `updateById`, there is no query-string. I will fix that

Reply:
Fixed